web/src/recent_view_ui.ts

```diff
@@ -373,12 +394,12 @@ export function revive_current_focus() {
         return true;
     }
 
-    if ($current_focus_elem.hasClass("dropdown-widget-button")) {
+    if ($current_focus_elem !== "table" && $current_focus_elem.hasClass("dropdown-widget-button")) {
```

Comment:
This check is not needed, we already checked that above. Could make sense to do an assert on the line after `is_table_focused` to keep using the semantic function for the main check, while letting TS know what that means for the type of `$current_focus_elem`.
```
function is_table_focused() {                                                                           
    return $current_focus_elem === "table";                                                             
}                                                                                                       
```

Reply:
I added `assert($current_focus_elem !== "table");` after the `is_table_focused` block, good catch!